src/components/ActiveTeamsComponent.jsx

```diff
@@ -11,84 +9,69 @@ const ActiveTeamsComponent = ({
-        )}
-        {teams.length > 0 && (
-          <ul ref={myListRef}>
-            {teams.map(
-              (team, index) =>
-                team.active && (
-                  <li key={index} draggable>
-                    <button
-                      type='button'
-                      onDragStart={handleDragStart(index)}
-                      onDragOver={handleDragOver}
-                    >
-                      Drag
-                    </button>
-                    <input
-                      type='text'
-                      value={team.name}
-                      onChange={(e) => handleNameInput(e, index)}
-                    />
-                    <button
-                      type='button'
-                      onClick={() => handleScoreDecrease(index)}
-                    >
-                      -
-                    </button>
-                    <span>{teams[index].score}</span>
-                    <button
-                      type='button'
-                      onClick={() => handleScoreIncrease(index)}
-                    >
-                      +
-                    </button>
-                    <button type='button' onClick={() => handleTransfer(index)}>
-                      Transfer
-                    </button>
-                    <button type='button' onClick={() => handleDelete(index)}>
-                      Delete
-                    </button>
-                  </li>
-                ),
-            )}
-          </ul>
-        )}
-      </div>
-      <button type='button' onClick={() => handleAddTeam('active')}>
-        Add team
-      </button>
+      {teams.length > 0 && (
+        <ul>
+          {teams.map(
+            (team) => team.active && (
+              <li key={team.id} draggable data-teamid={team.id}>
+                <button
+                  type="button"
+                  onDragStart={() => handleDragStart(team.id)}
+                  onDragOver={handleDragOver}
+                >
+                  Drag
+                </button>
+                <input
+                  type="text"
+                  value={team.name}
+                  onChange={handleNameInput}
+                />
+                <button
+                  type="button"
+                  onClick={handleScoreDecrease}
+                >
+                  -
+                </button>
+                <span>{team.score}</span>
+                <button
+                  type="button"
+                  onClick={handleScoreIncrease}
+                >
+                  +
+                </button>
+                <button type="button" onClick={handleTransfer}>
+                  Transfer
+                </button>
+                <button type="button" onClick={handleDelete}>
+                  Delete
+                </button>
+              </li>
+            )
+          )}
+        </ul>
+      )}
     </div>
-  );
-};
+    <button type="button" onClick={() => handleAddTeam('active')}>
```

Comment:
Same here. This anonymous function should be avoided. I had no time to refactor it, but the concept is the same.

Reply:
I will do that. Thanks.

Reply:
The difference here is passing a function as a reference vs calling a function. By passing as a reference it is not calles immediately. The reference tells (in this case the browser) to store the function for later execution and (in this case the browser) is calling the function on click event.

To not call it, you must omit the parenthesis. So
```suggestion
    <button type="button" onClick={handleAddTeam}>
```

But then you lose the attribute of course. So either use a different function for each use case, get the information of the active or eliminated team from somewhere else or use a trick where you combine two arrow functions in one. Here an unrelated example just for demonstration purposes and without useCallback to make it easier to understand:

```jsx
const handler = (active) => (event) => {
  if (active) {
    event.preventDefault();
  }
}
...
<button onClick={handler('active')}>Click me</button>
```

When **rendered**, the outer function gets called immediately and returns the **actual** callback function. By using this type of nesting with arrow functions, it's possible to pass down additional data to the actual handler itself.

Reply:
Okay, I thought of this actually, writing a wrapper function to replace the anonymous one. Noted. Thanks @iteratus